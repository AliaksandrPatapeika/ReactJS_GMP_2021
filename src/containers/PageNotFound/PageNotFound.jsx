import './PageNotFound.less';

import React from 'react';
import {Link} from 'react-router-dom';

import Button from '../../components/Button';
import errorPage from '../../img/errorPage.png';
import Footer from '../Footer';
import HeaderTop from '../HeaderTop';

const PageNotFound = () => (
  <>
    <div className="pageNotFoundContainer">
      <HeaderTop hideButton />
      <div className="pageNotFoundContent">
        <h1 className="title">Page Not Found</h1>
        <img
          src={errorPage}
          alt="page not found"
          className="errorPageImage"
        />
        <Link to="/">
          <Button className="goBackButton" title="GO BACK TO HOME" />
        </Link>
      </div>
    </div>
    <Footer />
  </>
);

export default PageNotFound;
