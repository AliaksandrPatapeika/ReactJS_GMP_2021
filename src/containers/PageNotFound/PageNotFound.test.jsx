import {head} from 'lodash';
import React from 'react';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import MockMovies from '../../tests/mocks/mockData';
import PageNotFound from './PageNotFound';

describe('PageNotFound', () => {
  test('renders correctly', () => {
    const mockMovie = head(MockMovies);
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        isLoading: true,
        movies: MockMovies,
        totalAmount: 6,
        activeModalWindow: 'addMovie',
        activeModalMovie: mockMovie
      },
      query: {
        filter: '',
        sortBy: 'release_date',
        sortOrder: 'desc'
      }
    };
    const store = mockStore(initialState);
    const pageNotFound = renderer
      .create(
        <Provider store={store}>
          <Router>
            <PageNotFound />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(pageNotFound).toMatchSnapshot();
  });
});
