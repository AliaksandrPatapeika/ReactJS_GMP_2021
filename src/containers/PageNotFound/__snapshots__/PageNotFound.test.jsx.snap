// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound renders correctly 1`] = `
Array [
  <div
    className="pageNotFoundContainer"
  >
    <div
      className="headerTopContainer"
    >
      <div
        className="logoContainer"
      >
        <span
          className="netflix"
        >
          netflix
        </span>
        <span
          className="roulette"
        >
          roulette
        </span>
      </div>
    </div>
    <div
      className="pageNotFoundContent"
    >
      <h1
        className="title"
      >
        Page Not Found
      </h1>
      <img
        alt="page not found"
        className="errorPageImage"
        src="test-file-stub"
      />
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="goBackButton"
          disabled={false}
          id={null}
          onClick={[Function]}
          type="button"
        >
          GO BACK TO HOME
          
        </button>
      </a>
    </div>
  </div>,
  <footer
    className="footerContainer"
  >
    <div
      className="logoContainer"
    >
      <span
        className="netflix"
      >
        netflix
      </span>
      <span
        className="roulette"
      >
        roulette
      </span>
    </div>
  </footer>,
]
`;
